server/services/v1/request_metadata.go

```diff
@@ -0,0 +1,101 @@
+package v1
+
+import (
+	"context"
+
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+type RequestMetadataCtxKey struct {
+}
+
+type AccessToken struct {
+	Namespace string
+	Sub       string
+}
+
+type RequestMetadata struct {
+	accessToken AccessToken
+	namespace   metadata.Namespace
+}
+
+// NamespaceExtractor - extract the namespace from context
+type NamespaceExtractor interface {
+	Extract(ctx context.Context) (metadata.Namespace, error)
```

Comment:
Do we need to return the object, or should we just Set/Get the namespace name? That should be enough for us to get the tenant in sessionMgr.

Reply:
updated.